crates/symbolicator/src/services/download/http.rs

```diff
@@ -84,21 +95,36 @@ impl HttpDownloader {
             Ok(response) => {
                 if response.status().is_success() {
                     log::trace!("Success hitting {}", download_url);
+
+                    let timeout = content_length_timeout(
+                        response
+                            .headers()
+                            .get(header::CONTENT_LENGTH)
+                            .and_then(|hv| hv.to_str().ok())
+                            .and_then(|s| s.parse::<u32>().ok()),
+                        self.streaming_timeout,
+                        self.download_timeout,
+                    );
+
                     let stream = response.bytes_stream().map_err(DownloadError::Reqwest);
 
-                    super::download_stream(file_source, stream, destination).await
+                    tokio::time::timeout(
+                        timeout,
+                        super::download_stream(file_source, stream, destination),
```

Comment:
we should probably pass the timeout through to this function. same for all the other sources, since the other sources use this function as well to stream bytes.